### ISSUE TITLE
Problem to use segio in multi thread environment

### ISSUE THREAD

**Issue by keven4ever** [Statoil/segyio#331]
Hello dear segyio developer,

I am trying to use python concurrent.futures to parse and dump each trace in segy in parallel. However I got exception `Can't pickle <class '_segyio.segyfd'> : import of module '_segyio' failed`. Do you happen to know any workaround for this?

**Comment by jokva** [comment]
Segyio is not pickle-able, and not built for threading.

**Comment by jokva** [comment]
Can you open multiple handles, one per file? If you're only reading that should be fine.

**Comment by keven4ever** [comment]
I might be able tofind a way without pickle segio in a python multiple thread environment. 
However, considering number of segy files we need to handle, i really would like to use spark with segyio, do you think i will have same pickle pb when use segyio in spark executor?

**Comment by keven4ever** [comment]
Ok, i will try it out and let you know

> Maybe, I haven't used spark personally. It will probably be just fine.

Ok, i will try it out and let you know

**Comment by jokva** [comment]
Splendid. If you run into problems, please open a new issue.

**Comment by keven4ever** [comment]
Ok, it works fine with spark.

**Comment by jokva** [comment]
Splendid, glad to hear that.
